gordon/interfaces.py

```diff
@@ -0,0 +1,86 @@
+from zope.interface import Attribute
+from zope.interface import Interface
+
+
+class IEventMessage(Interface):
+    id = Attribute('Unique message id')
+    phase = Attribute('Current phase of message')
+    data = Attribute('DNS change data')
+    history_log = Attribute('Log of actions performed on the message')
+
+    def __call__(id, data, phase=None):
+        """Initialize an EventMessage.
+
+        Args:
+            id (int): Unique message identifier.
```

Comment:
Yeah, I think this can wait until v2 so we can tackle this once we get there :). From a quick glance, it looks like different message queues use different id schemes (random, sequential, long, short, etc) so the chance is low, but it's there.

This also made me realize that a message id doesn't have to be an integer, it could be an alphanumeric string, so I changed the type of `msg_id` to `str` (from `int`).